ReproducibleEnvs.Rmd

```diff
@@ -0,0 +1,564 @@
+
+Including the packages used, **all the code is saved in context and documented with comments in scripts**. This also gives benefits such as syntax highlighting, diagnostics, auto-completion of functions, etc.
+
+To support this way of working, changing the RStudio Global Options to not save or load previous .RData files (your workspace) will encourage you to work with a reproducible mindset and write code in scripts.
+
+```{r renv-globalops, echo=FALSE, fig.align='center'}
+knitr::include_graphics("images/renv-globalops.jpg")
+```
+
+*When using PHS Posit Workbench, your R session doesn't end or restart when you close your browser as it's hosted on a server. It's important to consider this in terms of your workflow for managing your R workspace and saving elements but also as a use of resources within the organisation. See [Best Practice with R in Posit Workbench](https://public-health-scotland.github.io/knowledge-base/docs/Posit%20Infrastructure?doc=Best%20Practice%20with%20R%20in%20Posit%20Workbench.md) guidance.*
+
+From this, we can layer other best practice:
+
+-   Follow [PHS R Style Guide](https://public-health-scotland.github.io/knowledge-base/docs/R?doc=R%20Style%20Guide.md)
+-   File (script) names should be meaningful, machine- and human-readable.
+-   Files should be ordered within a single project directory, provided with leading numbers for ordered lists.
+-   Outputs are stored in a set location and excluded from version control.
+
+</br>
+
+#### Projects
+
+The next major component of setting up a project for reproducibility is using RStudio Projects. R accesses a working directory, found using `getwd()`, and can be changed using `setwd()` (not recommended). This is likely to change between users but also between your projects which could cause confusion and issue when switching from one bit of work to another. This can be set up as below from File > New Project (we'll come back to setting up projects with renv - the green box):
+
+```{r renv-newproj, echo=FALSE, fig.align='center'}
+knitr::include_graphics("images/renv-newproj1.png")
+```
+
+Building on the solid foundations that RStudio Projects provide, we need to use relative paths within our code. A relative path is *relative* to the current working directory (set by the project). This means that if an output is saved using a relative path: `data/project_data.csv`, different users would access the same location regardless of where their working version of the project (i.e. `project`) is stored, e.g.:
+
+* `/Projects/user1/project/data/project_data.csv`
+* `/Projects/user2/phs_projects/project/data/project_data.csv`
+
+While less of an issue in PHS as we all use Posit Workbench to access RStudio, there can also be differences introduced depending on operating systems. If you're working with external collaborators for instance, they may use a Windows system which would require different formatting for file paths. 
```

Comment:
Possibly worth including here, that even within PHS Posit WB we can have different versions of R - 3 at the moment and these are likely (guaranteed?) to change / increase over time.